### PR TITLE
Fix/tab groups

### PR DESCRIPTION
**Pull request by bjagg**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [ ] documentation is changed or added
-   [x] view conforms with [WCAG 2.0 AA][]

##### Description of change

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by bjagg** [comment]
[campus-lo.fragment-layout copy.txt](https://github.com/Jasig/uPortal/files/5799037/campus-lo.fragment-layout.copy.txt)

**Comment by bjagg** [comment]
[guest-lo.fragment-layout copy.txt](https://github.com/Jasig/uPortal/files/5799145/guest-lo.fragment-layout.copy.txt)

File: uPortal-webapp/src/main/resources/layout/theme/respondr/navigation.xsl [hunk 1]
```diff
@@ -89,6 +89,54 @@
...
+                    <div id="portalNavigationTabGroup" class="fl-widget-inner header">
+                        <!-- <span id="activeTabGroup" style="display:none;"><xsl:value-of select="/layout/navigation/tabGroupsList/@activeTabGroup"/></span> -->
+                        <ul id="portalNavigationTabGroupsList" class="menu fl-tabs list-group list-group-horizontal">
+                            <xsl:for-each select="tabGroupsList/tabGroup">
+                                <xsl:variable name="TABGROUP_POSITION"> <!-- Determine the position of the navigation option within the whole navigation list and add css hooks for the first and last positions. -->
+                                    <xsl:choose>
+                                        <xsl:when test="position()=1 and position()=last()">single singleTabGroup</xsl:when>
+                                        <xsl:when test="position()=1">first</xsl:when>
+                                        <xsl:when test="position()=last()">last</xsl:when>
+                                        <xsl:otherwise></xsl:otherwise>
```
Comment (ChristianMurphy) [inline uPortal-webapp/src/main/resources/layout/theme/respondr/navigation.xsl:108]: Is this CSS we control?
Could CSS [`:first-child`](https://developer.mozilla.org/en-US/docs/Web/CSS/:first-child) and [`:last-child`](https://developer.mozilla.org/en-US/docs/Web/CSS/:last-child) be used?

File: uPortal-webapp/src/main/webapp/media/skins/respondr/common/less/navigation.less [hunk 2]
```diff
@@ -390,6 +390,76 @@
           -o-transform: none !important;
           transform: none !important;
         }
+        div#portalNavigationTabGroup {
```
Comment (ChristianMurphy) [inline uPortal-webapp/src/main/webapp/media/skins/respondr/common/less/navigation.less:393]: is there only one `portalNavigationTabGroup`?
If so
```suggestion
        #portalNavigationTabGroup {
```

would be unique enough

File: uPortal-webapp/src/main/webapp/media/skins/respondr/common/less/navigation.less [hunk 3]
```diff
@@ -390,6 +390,76 @@
...
+
+                &:hover,
+                &:focus {
+                  background-color: transparent;
+                  color: @tabgroup-item-hover-text-color;
+                }
+              }
+
+              a {
+              }
```
Comment (ChristianMurphy) [inline uPortal-webapp/src/main/webapp/media/skins/respondr/common/less/navigation.less:449]: unused group?
```suggestion
```

**Review by ChristianMurphy — APPROVED** [review]

File: uPortal-webapp/src/main/resources/layout/theme/respondr/navigation.xsl (hunk 1 above)
Comment (ChristianMurphy) [reply uPortal-webapp/src/main/resources/layout/theme/respondr/navigation.xsl:108]: In theory

```css
.portal-navigation-tabgroup:first-child {} /* in lieu of `.portal-navigation-tabgroup.first` */
.portal-navigation-tabgroup:last-child  {} /* in lieu of `.portal-navigation-tabgroup.last` */
.portal-navigation-tabgroup:only-child  {} /* in lieu of `.portal-navigation-tabgroup.singleTabGroup` */
```

could be used

Comment (cbeach47) [reply uPortal-webapp/src/main/resources/layout/theme/respondr/navigation.xsl:108]: looks like it'll work.

File: uPortal-webapp/src/main/webapp/media/skins/respondr/common/less/navigation.less [hunk 4]
```diff
@@ -390,6 +390,74 @@
...
         }
+        #portalNavigationTabGroup {
+          background-color: @tabgroup-background-color;
+          margin-left: -15px;
+          padding-left: 15px;
+          margin-right: -15px;
+          padding-right: 15px;
+
+          #portalNavigationTabGroupsList {
+            //display: block;
```
Comment (ChristianMurphy) [inline uPortal-webapp/src/main/webapp/media/skins/respondr/common/less/navigation.less:401]:
```suggestion
```

File: uPortal-webapp/src/main/webapp/media/skins/respondr/common/less/navigation.less [hunk 5]
```diff
@@ -390,6 +390,74 @@
...
+
+              &:last-child {
+                border-bottom: initial;
+              }
+
+              a.portal-navigation-link {
+                display: inline-block;
+                color: @tabgroup-item-text-color;
+
+                /* line-height: 30px; */
```
Comment (ChristianMurphy) [inline uPortal-webapp/src/main/webapp/media/skins/respondr/common/less/navigation.less:437]:
```suggestion
```

File: uPortal-webapp/src/main/webapp/media/skins/respondr/common/less/navigation.less [hunk 6]
```diff
@@ -390,6 +390,74 @@
...
+                  color: @tabgroup-item-hover-text-color;
+                }
+              }
+
+            }
+
+            .active {
+              background-color: @tabgroup-item-active-background-color;
+              color: @tabgroup-item-active-text-color;
+              &:hover,
```
Comment (ChristianMurphy) [inline uPortal-webapp/src/main/webapp/media/skins/respondr/common/less/navigation.less:453]:
```suggestion
              color: @tabgroup-item-active-text-color;

              &:hover,
```

File: uPortal-webapp/src/main/resources/layout/theme/respondr/navigation.xsl [hunk 7]
```diff
@@ -89,6 +89,46 @@
       <nav class="portal-nav">
         <div id="sidebar" class="sidebar-offcanvas container-fluid">
             <div id="portalNavigation" class="fl-widget" aria-selected="false">
+                <!-- Optional Tab Groups -->
+                <xsl:if test="$USE_TAB_GROUPS='true'">
+                    <!--
+                    <span id="activeTabGroup" style="display:none;"><xsl:value-of select="/layout/navigation/tabGroupsList/@activeTabGroup"/></span>
+                    <span id="useTabGroup" style="display:none;"><xsl:value-of select="$USE_TAB_GROUPS"/></span>
+                    <h3 class="fl-offScreen-hidden">Tab Groups</h3>
+                    -->
```
Comment (ChristianMurphy) [inline uPortal-webapp/src/main/resources/layout/theme/respondr/navigation.xsl:98]:
```suggestion
```

File: uPortal-webapp/src/main/resources/layout/theme/respondr/navigation.xsl [hunk 8]
```diff
@@ -89,6 +89,46 @@
...
+                    <div id="portalNavigationTabGroup" class="fl-widget-inner header">
+                        <!-- <span id="activeTabGroup" style="display:none;"><xsl:value-of select="/layout/navigation/tabGroupsList/@activeTabGroup"/></span> -->
```
Comment (ChristianMurphy) [inline uPortal-webapp/src/main/resources/layout/theme/respondr/navigation.xsl:100]:
```suggestion
```

**Review by ChristianMurphy — APPROVED** [review]

**Comment by cbeach47** [comment]
Looks good :+1:   taking for a final test

**Review by ChristianMurphy — APPROVED** [review]
Thanks @bjagg and @cbeach47! :bow:

**Comment by cbeach47** [comment]
.use-tabs-menu-mixin > #wrapper > .portal-nav > sidebar-offcanvas > #portalNavigationTabGroup > #portalNavigationTabGroupsList > .list-group-item > 
  &:last-child {
    background-color: green;              
  }
 &:first-child {
   background-color: yellow;
  }
  &:only-child {
    background-color: red;
  }
```

**Review by cbeach47 — APPROVED** [review]
Tested on FF, Chrome, IE, Safari, Edge with guest (multiple tabs), admin (no tabs), and staff (multiple tabs).  Tested in FF mobile view.  

Looks good :+1:

**Pull request merged**